grant-project/grant-monorepo
# Email verification

Closes #154.

### What This Does

Adds email verification to the spec of #154. In short:

* Adds `email_verification` table with userid, code, and has_verified columns
    * 1:1 relationship with users table, an email_verification is created when user is created
* Adds POST endpoint `/email/<code>/verify` that marks has_verified as true
* Adds a page at `/email/verify?code={code}` which verifies the email
* Minor style updates to email template

Verification currently does nothing yet, but this is good scaffolding for the future where we may require verification to do things like comment or make a proposal.

### Steps to Test

Make sure you've run `flask db upgrade`. May be problematic if you've run different migrations on another branch. You'll also need to setup the sendgrid API and SITE_URL environment variables or else you won't get emails.

* Create a new account, confirm you got an email
* Open the email, confirm the link(s) point to the right place
* Click the link, confirm email verifies correctly

## Screenshots

### Email

<img width="735" alt="screen shot 2018-11-01 at 5 21 24 pm" src="https://user-images.githubusercontent.com/649992/47880808-9f5c7800-ddfa-11e8-904f-cae220389f0e.png">

### Verified

<img width="952" alt="screen shot 2018-11-01 at 5 21 37 pm" src="https://user-images.githubusercontent.com/649992/47880812-a3889580-ddfa-11e8-94cc-77312d71d136.png">
